tuxemon/item/effects/evolve.py

```diff
@@ -0,0 +1,54 @@
+#
+# Tuxemon
+# Copyright (c) 2014-2017 William Edwards <shadowapex@gmail.com>,
+#                         Benjamin Bean <superman2k5@gmail.com>
+#
+# This file is part of Tuxemon
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Contributor(s):
+# Adam Chevalier <chevalieradam2@gmail.com>
+#
+
+from __future__ import annotations
+
+from typing import NamedTuple
+
+from tuxemon.item.itemeffect import ItemEffect, ItemEffectResult
+from tuxemon.monster import Monster
+
+
+class EvolveEffectResult(ItemEffectResult):
+    pass
+
+
+class EvolveEffectParameters(NamedTuple):
+    monster_evolve: str
+
+
+class EvolveEffect(ItemEffect[EvolveEffectParameters]):
+    """This effect evolves the target into the monster in the parameters."""
+
+    name = "evolve"
+    param_class = EvolveEffectParameters
+
+    def apply(self, target: Monster) -> EvolveEffectResult:
+        monster_evolve = self.parameters.monster_evolve
+
+        if any(d["monster_slug"] == monster_evolve for d in target.evolutions):
+            self.user.evolve_monster(target,monster_evolve)
```

Comment:
run black and isort on all files you modified

Reply:
There's a formatting script in the scripts directory, there will be a few changes as people haven't been running it.

Run this against the developer branch and then raise a PR.

When that is merged in you can rebase this branch and run the script again against your current changes.

Reply:
No

Reply:
Ok, just run black and isort against the files you have added / modified. There's only 5 of them.